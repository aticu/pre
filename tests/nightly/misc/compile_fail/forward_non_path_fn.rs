use pre::pre;

fn foo() {}

#[pre]
fn main() {
    let array = [foo];

    #[forward(foo)]
    array[0]();
}
