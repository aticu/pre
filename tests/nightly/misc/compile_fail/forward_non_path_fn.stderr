error: unable to determine at compile time which function is being called
  --> $DIR/forward_non_path_fn.rs:10:5
   |
10 |     array[0]();
   |     ^^^^^^^^
   |
   = help: use a direct path to the function instead
